### ISSUE TITLE
Accessibility violations for checkboxes

### ISSUE THREAD

**Issue by dszewczyk1** [surveyjs/survey-library#3772]
### Are you requesting a feature, reporting a bug or asking a question?
Question/bug

### What is the current behavior?
Violation reports based on IBM Equal Access Accessibility Checker (Chrome extension) https://chrome.google.com/webstore/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp)
> `<fieldset>` element `<legend>` is empty
> An element with ARIA role 'checkbox' does not have the required ARIA attribute(s): 'aria-checked'
> The ARIA role 'checkbox' is not valid for the element `<legend>`

And that's because of the `<legend>` added like this:
`<legend role="checkbox" aria-label="What car are you driving?"></legend>`

I can see the reasoning that you've added aria-label instead of providing a text (in other way, you'd need to visually hide it) 
But why `<legend>` even has role="checkbox"?

It's also complaining about
> Accessible name does not match or contain the visible label text

I guess it doesn't like the aria-label here, which doesn't match "Ford"
`<input value="Ford" aria-invalid="false" aria-label="What car are you driving?" aria-required="true" id="sq_100i_0" name="car" type="checkbox" class="sv-visuallyhidden sv-item__control">`

### What is the expected behavior?
No violations reported

### How would you reproduce the current behavior (if this is a bug)?
Run IBM Equal Access Accessibility Checker scan on a page with checkboxes

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: https://surveyjs.io/Examples/Library?id=questiontype-checkbox&platform=Reactjs&theme=modern

### Specify your

* browser: Chrome
* browser version: 96.0.4664.93
* surveyjs platform (angular or react or jquery or knockout or vue): all, I think (but React in my case)
* surveyjs version: 1.9.2

**Comment by dmitrykurmanov** [comment]
We will investigate the problem and write back asap

**Comment by dmitrykurmanov** [comment]
I believe that we fixed the most problems here. At least nvda reader sounds ok now. Fix will be available in the next minor release.

As for `fieldset` and `legend` tags we don't use them for the accessibility because of we have a very custom and complex markup for the questions. We set `role="presentation"` for them (IBM checker doesn't respect the `role="presentation"` for some reasons). That's why we use custom `aria-labels`.

**Comment by dszewczyk1** [comment]
@dmitrykurmanov @tsv2013 I can understand that you have different use cases for questions layout, but in regards to `fieldset` it seems pretty straightforward and something that would be expected, that it contains a `legend` field, which describes it (it could have some kind of `sr-only` class to have it visually hidden).
Perhaps in terms of a11y setting fieldset's role to `presentation` doesn't change this requirement to still have a `legend`.

I checked the changes and also there's a new thing raised by this checker. 
`The ARIA attribute 'aria-required, aria-invalid' is not valid for the element <div> with ARIA role 'group'`
https://www.w3.org/TR/wai-aria-1.2/#group

I know that on one hand it still works correctly in screen reader, but on the other hand testers/clients are just using a tool and seeing a violation...

**Comment by dmitrykurmanov** [comment]
@dszewczyk1, probably we should replace `fieldset` with  the `div` because of our very custom layout. We will discuss it with the team and return back here.
